src/sqlfluff/cli/formatters.py

```diff
@@ -252,7 +252,10 @@ def _format_config(self, linter: Linter) -> str:
                 ("sqlfluff", get_package_version()),
                 ("python", get_python_version()),
                 ("implementation", get_python_implementation()),
-                ("dialect", linter.dialect.name),
+            ]
+            if linter.dialect:
+                config_content.append(("dialect", linter.dialect.name))
+            config_content += [
                 ("verbosity", self._verbosity),
             ] + linter.templater.config_pairs()
```

Comment:
Should we move this up to the previous section to make this a little more readable:

```py
            config_content = [
                ("sqlfluff", get_package_version()),
                ("python", get_python_version()),
                ("implementation", get_python_implementation()),
                ("dialect", linter.dialect.name),
                ("verbosity", self._verbosity),
            ]
            if linter.dialect:
                config_content.append(("dialect", linter.dialect.name))
            config_content += linter.templater.config_pairs()
```

Reply:
We could. I was trying to keep the same order, in case it broke some tests, but I'll give it a try.